internal/controllers/control/cassandratask_controller.go

```diff
@@ -815,6 +814,16 @@ func (r *CassandraTaskReconciler) reconcileEveryPodTask(ctx context.Context, dc
 					<-jobRunner
 				}()
 
+				if taskConfig.SyncFeature != "" {
+					if !features.Supports(taskConfig.SyncFeature) {
+						logger.Error(err, "Pod doesn't support this feature", "Pod", pod, "Feature", taskConfig.SyncFeature)
+						jobStatus.Status = podJobError
+						failed++
+						errMsg = fmt.Sprintf("Pod %s doesn't support %s feature", pod.Name, taskConfig.SyncFeature)
+						return
```

Comment:
This is a bit trickier to implement as this state is not persisted now at all. Remember, we're inside a goroutine so nothing outside is usable. So failed++ is not doing anything and nor is returning here (as it terminates the goroutine). The jobStatus is stored on the pod and that's done on the line 848 so we need to jump there. 

I wonder if you intended this check to happen at line 779 with similar behavior as the case "if taskConfig.SyncFunc == nil {" ?

Reply:
Makes sense, fixed...